src/main/java/org/trustedanalytics/h2oscoringengine/h2omodel/H2oModel.java

```diff
@@ -17,22 +17,66 @@
 import static com.google.common.base.Preconditions.checkArgument;
 
 import hex.genmodel.GenModel;
+import hex.genmodel.easy.RowData;
+import hex.genmodel.easy.EasyPredictModelWrapper;
+import hex.genmodel.easy.prediction.*;
+import hex.genmodel.easy.exception.*;
+
+import java.util.HashMap;
+import java.util.Arrays;
 
 public class H2oModel {
-    
+
     private final GenModel model;
+    private final String[] names;
 
     public H2oModel(GenModel model) {
         this.model = model;
+        this.names = model.getNames();
     }
 
-    public double[] score(double[] data) {
+    public String[] score(HashMap<String, Object> data) {
+
+        RowData row = new RowData();
+        for (HashMap.Entry<String, Object> entry : data.entrySet())
+        {
+              String key = entry.getKey();
+              String val = entry.getValue().toString();
+              if (val != "" && Arrays.asList(this.names).contains(key)) {
+                  row.put(key, val);
+              }
+        }
+
+        EasyPredictModelWrapper predictor = new EasyPredictModelWrapper(model);
+
+        try {
+          BinomialModelPrediction pred = predictor.predictBinomial(row);
+          String[] result = {pred.label, String.format("%.5f", pred.classProbabilities[0]), String.format("%.5f", pred.classProbabilities[1])};
+          return result;
+        }
+        catch (PredictUnknownCategoricalLevelException e) {
+          data.remove(e.getColumnName());
+          return this.score(data);
```

Comment:
This exception means that the wrong input was provided. It would be better not to try to fix the input, but simply return Bad Request response to the client, with a column name and a message, that there was no such column during model training.